edb/pgsql/compiler/relgen.py

```diff
@@ -1346,22 +1349,25 @@ def process_set_as_subquery(
                 set_rvar, ir_set.path_id, env=ctx.env)
 
             pathctx.get_path_identity_output(
-                stmt, ir_set.path_id, env=ctx.env)
-            cond_expr = astutils.new_binop(tgt_ref, stmt, 'IN')
+                ctx.rel, ir_set.path_id, env=ctx.env
+            )
+            cond_expr = astutils.new_binop(tgt_ref, ctx.rel, 'IN')
 
             # Make a new stmt, join in the new root, and semi join on
             # the original statement.
-            stmt = pgast.SelectStmt()
-            relctx.include_rvar(stmt, set_rvar, ir_set.path_id, ctx=ctx)
-            stmt.where_clause = astutils.extend_binop(
-                stmt.where_clause, cond_expr)
+            ctx.rel = pgast.SelectStmt()
```

Comment:
This is probably wrong (even if doesn't cause evident trouble), because you are not supposed to mutate a context that you don't own.  You can avoid this by keeping the `stmt` arg in `_new_subquery_stmt_set_rvar`